lib/elixir/lib/registry.ex

```diff
@@ -1170,7 +1173,13 @@ defmodule Registry do
   @doc since: "1.x.x"
   @spec select(registry, spec) :: [term]
   def select(registry, spec) do
-    spec = to_internal_spec(spec)
+    spec =
+      for part <- spec do
+        case part do
+          {{key, pid, value}, guards, select} -> {{key, {pid, value}}, guards, select}
+          part -> part
```

Comment:
We should raise on this branch. :)

Reply:
Hmm, originally I wanted to support `:_` which `:ets.select` does but considering we're recommending users not to use `:"$_"` in the body part it doesn't necessarily make sense to allow `:_` in the match part. I can switch it to raise an argumenterror. We could have a branch  with a guard to  check that the first part of the tuple is an atom and raise in all other cases, but  again, it might not make sense to allow `:_`